README.md

```diff
@@ -93,7 +93,39 @@ translator.jsonapify(params)
 
 As [stated in the JSON:API specification](https://jsonapi.org/#mime-types) correct mime type for JSON:API input should be [`application/vnd.api+json`](http://www.iana.org/assignments/media-types/application/vnd.api+json). 
 
-This gems intention is to make input consumption as easy as possible. Hence, it [registers this mime type for you](lib/jsonapi_parameters/core_ext/action_dispatch/http/mime_type.rb). 
+This gems intention is to make input consumption as easy as possible. Hence, it [registers this mime type for you](lib/jsonapi_parameters/core_ext/action_dispatch/http/mime_type.rb).
+
+## Stack limit
+
+In theory, any payload may consist of infinite amount of relationships (and so each relationship may have its own, included, infinite amount of nested relationships).
+Because of that, it is a potential vector of attack. 
+
+For this reason we have introduced a default limit of stack levels that JsonApi::Parameters will go down through while parsing the payloads. 
+
+This default limit is 3, and can be overwritten by specifying the custom limit.
+
+#### Ruby
+```
+class Translator
+    include JsonApi::Parameters
+end
+
+translator = Translator.new
+translator.stack_limit = 4
+``` 
+
+#### Rails
+```
+# config/initializers/jsonapi_parameters.rb
+
+def create_params
+    params.stack_level = 4
+
+    params.from_jsonapi.require(:user).permit(entities_attributes: { subentities_attributes: { ... } })
+ensure
+    params.reset_stack_limit!
+end
```

Comment:
It is not required to have `reset_stack_limit!` called each time, as `#stack_level=` setter only changes the stack level limitation in that given instance of ActionController::Parameters (or whatever structure was `JsonApi::Parameters` included into). 

The suggestion of allowing `#from_jsonapi` to receive an optional argument of stack level limitation is cool, it could actually just do `stack_limit = arg` behind the scene.